src/components/organisms/shared/OakPagination/OakPagination.tsx

```diff
@@ -137,20 +140,53 @@ export const OakPagination = ({
   prevHref,
   paginationHref,
   pageName,
+  firstItemRef,
+  router,
+  shouldAppendQuery,
 }: OakPaginationProps) => {
+  useEffect(() => {
+    if (router.query.page && firstItemRef?.current) {
+      firstItemRef.current.focus();
+      window.scrollTo({ top: 0, behavior: "smooth" });
+    }
+  }, [firstItemRef, router.query.page]);
+
   const [activePage, setActivePage] = useState(currentPage);
 
   const pages = generatePageNumbers(activePage, totalPages);
 
   const isFirstPage = activePage <= 1;
   const isLastPage = activePage >= totalPages;
 
-  const handleNumberClick = (num: number) => {
+  const handleNumberClick = (num: number, event: React.MouseEvent) => {
+    event.preventDefault();
     setActivePage(num);
+
+    shouldAppendQuery
+      ? router.push(`${paginationHref}&page=${num}`, undefined, {
```

Comment:
when is `shouldAppendQuery` used? can we use a URL object to include all required query params instead? https://nextjs.org/docs/pages/api-reference/functions/use-router#with-url-object

Reply:
Removed this so OWA sets URL ✅